src/main/java/hudson/plugins/plot/pipeline/PlotPublisher.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright (c) 2007-2009 Yahoo! Inc.  All rights reserved.
+ * The copyrights to the contents of this file are licensed under the MIT License (http://www.opensource.org/licenses/mit-license.php)
+ */
+package hudson.plugins.plot.pipeline;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.Util;
+import hudson.matrix.MatrixProject;
+import hudson.model.AbstractProject;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.plot.AbstractPlotPublisher;
+import hudson.plugins.plot.Messages;
+import hudson.plugins.plot.Plot;
+import hudson.plugins.plot.SeriesFactory;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.BuildStepMonitor;
+import hudson.tasks.Publisher;
+import hudson.tasks.Recorder;
+import hudson.util.FormValidation;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import javax.annotation.Nonnull;
+import jenkins.tasks.SimpleBuildStep;
+import net.sf.json.JSONObject;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerRequest;
+
+/**
+ * Records the plot data for builds.
+ *
+ * @author Nigel Daley
+ */
+public class PlotPublisher extends Recorder implements SimpleBuildStep {
```

Comment:
🐛  Why a new class?  Why not leave where it was and ensure you're not breaking compatibility for existing projects with a PlotPublisher configured?

Reply:
I'm assuming this is mostly copy+pasted from the old Impl, so I'm only giving it a quick skim through.  Will look closer if that's not true.